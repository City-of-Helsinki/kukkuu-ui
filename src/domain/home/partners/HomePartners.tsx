import * as React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './homePartners.module.scss';
import Icon from '../../../common/components/icon/Icon';
import amosrexIcon from '../../../assets/icons/partners/amosrex.png';
import cirkoIcon from '../../../assets/icons/partners/cirko.png';
import designmuseumIcon from '../../../assets/icons/partners/designmuseum.png';
import hamhelsinkiIcon from '../../../assets/icons/partners/hamhelsinki.png';
import helIcon from '../../../assets/icons/partners/hel.png';
import helsinginkaupunginmuseoIcon from '../../../assets/icons/partners/helsinginkaupunginmuseo.png';
import helsinginkaupunginorkesteriIcon from '../../../assets/icons/partners/helsinginkaupunginorkesteri.png';
import hktIcon from '../../../assets/icons/partners/hkt.png';
import hotellijaravintolamuseoIcon from '../../../assets/icons/partners/hotellijaravintolamuseo.png';
import hurjaruuthIcon from '../../../assets/icons/partners/hurjaruuth.png';
import jaesIcon from '../../../assets/icons/partners/jaes.png';
import kansallismuseoIcon from '../../../assets/icons/partners/kansallismuseo.png';
import kansallisteatteriIcon from '../../../assets/icons/partners/kansallisteatteri.png';
import kaviIcon from '../../../assets/icons/partners/kavi.png';
import kulttuuriperintokasvatusIcon from '../../../assets/icons/partners/kulttuuriperintokasvatus.png';
import mfaIcon from '../../../assets/icons/partners/mfa.png';
import nukketeatterisampoIcon from '../../../assets/icons/partners/nukketeatterisampo.png';
import oodihelsinkiIcon from '../../../assets/icons/partners/oodihelsinki.png';
import oopperabalettiIcon from '../../../assets/icons/partners/oopperabaletti.png';
import qteatteriIcon from '../../../assets/icons/partners/qteatteri.png';
import svenskateaternIcon from '../../../assets/icons/partners/svenskateatern.png';
import tanssintaloIcon from '../../../assets/icons/partners/tanssintalo.png';
import teatteriilmioIcon from '../../../assets/icons/partners/teatteriilmio.png';
import teatterimuseoIcon from '../../../assets/icons/partners/teatterimuseo.png';
import valokuvataiteenmuseoIcon from '../../../assets/icons/partners/valokuvataiteenmuseo.png';

const HomePartners: React.FunctionComponent = props => {
  const { t } = useTranslation();

  interface Partner {
    name: string;
    icon: string;
  }

  type Props = {
    partners: Partner[];
  };

  // Name, icon file name and translation key is created from partner's domain
  // name without .fi and dash.
  // Examples:
  // teatteri-ilmio.fi -> teatteriilmio
  // hel.fi -> hel
  const mainPartnerList: Partner[] = [
    { name: 'hel', icon: helIcon },
    { name: 'jaes', icon: jaesIcon },
  ];
  const partnerList: Partner[] = [
    { name: 'amosrex', icon: amosrexIcon },
    { name: 'cirko', icon: cirkoIcon },
    { name: 'designmuseum', icon: designmuseumIcon },
    { name: 'hamhelsinki', icon: hamhelsinkiIcon },
    { name: 'helsinginkaupunginmuseo', icon: helsinginkaupunginmuseoIcon },
    {
      name: 'helsinginkaupunginorkesteri',
      icon: helsinginkaupunginorkesteriIcon,
    },
    { name: 'hkt', icon: hktIcon },
    { name: 'hotellijaravintolamuseo', icon: hotellijaravintolamuseoIcon },
    { name: 'hurjaruuth', icon: hurjaruuthIcon },
    { name: 'kansallismuseo', icon: kansallismuseoIcon },
    { name: 'kansallisteatteri', icon: kansallisteatteriIcon },
    { name: 'kavi', icon: kaviIcon },
    { name: 'kulttuuriperintokasvatus', icon: kulttuuriperintokasvatusIcon },
    { name: 'mfa', icon: mfaIcon },
    { name: 'nukketeatterisampo', icon: nukketeatterisampoIcon },
    { name: 'oodihelsinki', icon: oodihelsinkiIcon },
    { name: 'oopperabaletti', icon: oopperabalettiIcon },
    { name: 'qteatteri', icon: qteatteriIcon },
    { name: 'svenskateatern', icon: svenskateaternIcon },
    { name: 'tanssintalo', icon: tanssintaloIcon },
    { name: 'teatteriilmio', icon: teatteriilmioIcon },
    { name: 'teatterimuseo', icon: teatterimuseoIcon },
    { name: 'valokuvataiteenmuseo', icon: valokuvataiteenmuseoIcon },
  ];

  function Partners(props: Props) {
    // any is workaround for TS bug
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const partners: any = props.partners;
    const partnerItems = partners.map((partner: Partner, index: number) => (
      <Icon
        key={index}
        className={styles.icon}
        src={partner.icon}
        alt={t(`home.partners.partner.${partner.name}`)}
      />
    ));
    return partnerItems;
  }

  return (
    <section className={styles.wrapper}>
      <h2>{t('home.partners.heading.text')}</h2>
      <div className={styles.partners}>
        <div className={styles.big}>
          <Partners partners={mainPartnerList} />
        </div>
        <div className={styles.small}>
          <Partners partners={partnerList} />
          <Icon
            className={styles.icon}
            src={helsinginkaupunginorkesteriIcon}
            alt=""
          />
        </div>
      </div>
    </section>
  );
};

export default HomePartners;
