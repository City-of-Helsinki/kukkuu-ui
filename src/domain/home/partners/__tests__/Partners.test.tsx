import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import HomePartners from '../HomePartners';

it('renders snapshot correctly', () => {
  const homePartners = shallow(<HomePartners />);
  expect(toJson(homePartners)).toMatchSnapshot();
});
