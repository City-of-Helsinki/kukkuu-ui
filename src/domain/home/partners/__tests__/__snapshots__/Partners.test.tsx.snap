// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<section
  className="wrapper"
>
  <h2>
    Yhteisty√∂kumppanit
  </h2>
  <div
    className="partners"
  >
    <div
      className="big"
    >
      <Partners
        partners={
          Array [
            Object {
              "icon": "hel.png",
              "name": "hel",
            },
            Object {
              "icon": "jaes.png",
              "name": "jaes",
            },
          ]
        }
      />
    </div>
    <div
      className="small"
    >
      <Partners
        partners={
          Array [
            Object {
              "icon": "amosrex.png",
              "name": "amosrex",
            },
            Object {
              "icon": "cirko.png",
              "name": "cirko",
            },
            Object {
              "icon": "designmuseum.png",
              "name": "designmuseum",
            },
            Object {
              "icon": "hamhelsinki.png",
              "name": "hamhelsinki",
            },
            Object {
              "icon": "helsinginkaupunginmuseo.png",
              "name": "helsinginkaupunginmuseo",
            },
            Object {
              "icon": "helsinginkaupunginorkesteri.png",
              "name": "helsinginkaupunginorkesteri",
            },
            Object {
              "icon": "hkt.png",
              "name": "hkt",
            },
            Object {
              "icon": "hotellijaravintolamuseo.png",
              "name": "hotellijaravintolamuseo",
            },
            Object {
              "icon": "hurjaruuth.png",
              "name": "hurjaruuth",
            },
            Object {
              "icon": "kansallismuseo.png",
              "name": "kansallismuseo",
            },
            Object {
              "icon": "kansallisteatteri.png",
              "name": "kansallisteatteri",
            },
            Object {
              "icon": "kavi.png",
              "name": "kavi",
            },
            Object {
              "icon": "kulttuuriperintokasvatus.png",
              "name": "kulttuuriperintokasvatus",
            },
            Object {
              "icon": "mfa.png",
              "name": "mfa",
            },
            Object {
              "icon": "nukketeatterisampo.png",
              "name": "nukketeatterisampo",
            },
            Object {
              "icon": "oodihelsinki.png",
              "name": "oodihelsinki",
            },
            Object {
              "icon": "oopperabaletti.png",
              "name": "oopperabaletti",
            },
            Object {
              "icon": "qteatteri.png",
              "name": "qteatteri",
            },
            Object {
              "icon": "svenskateatern.png",
              "name": "svenskateatern",
            },
            Object {
              "icon": "tanssintalo.png",
              "name": "tanssintalo",
            },
            Object {
              "icon": "teatteriilmio.png",
              "name": "teatteriilmio",
            },
            Object {
              "icon": "teatterimuseo.png",
              "name": "teatterimuseo",
            },
            Object {
              "icon": "valokuvataiteenmuseo.png",
              "name": "valokuvataiteenmuseo",
            },
          ]
        }
      />
      <Icon
        alt=""
        className="icon"
        src="helsinginkaupunginorkesteri.png"
      />
    </div>
  </div>
</section>
`;
